### ISSUE TITLE
null dereference when Animated view has a null child

### ISSUE THREAD

**Issue by chrisnojima** [software-mansion/react-native-reanimated#4129]
### Description

This check
```
export function isSharedValue<T>(value: any): value is SharedValue<T> {
  'worklet';
  return typeof value === 'object' && value._isReanimatedSharedValue === true;
}
```
https://github.com/software-mansion/react-native-reanimated/blob/a073f212e8e4e9c7c1e229d10fe5b21d3d980bef/src/reanimated2/utils.ts#L33

fails as `typeof null` is `object', then its immediately dereferenced so
```
<Animated.View>{maybeNull}</Animated.View> 
```

will crash out

### Steps to reproduce

Have an animated view around a maybe null child

### Snack or a link to a repository

N/A

### Reanimated version

3.0.0

### React Native version

0.70.7

### Platforms

iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by tomekzaw** [comment]
Should be fixed with #4128.

**Comment by TomCorvus** [comment]
@tomekzaw Do you know when these fixes will be merged and released?
Thanks
